src/Microsoft.PowerShell.Commands.Management/commands/management/GetContentCommand.cs

```diff
@@ -356,7 +352,7 @@ private bool ScanForwardsForTail(ContentHolder holder, CmdletProviderContext cur
                 {
                     while (tailResultQueue.Count >= ReadCount)
                     {
-                        ArrayList outputList = new ArrayList();
+                        ArrayList outputList = new ArrayList((int)ReadCount);
```

Comment:
Do you want `List<object>` here? Potentially faster because `ArrayList.Add` is virtual and `List<T>.Add` is not. Perhaps the JIT can detect that `outputList` always refers to the same type of object, and devirtualize the call anyway.

Reply:
It will a breaking change because of ArrayList converted to PSObject below.
If this make sense it should be in another PR.

Reply:
Where is it converted to PSObject? The `WriteContentObject` call uses `outputList.ToArray()`, rather than `outputList` itself.<https://github.com/PowerShell/PowerShell/blob/0e9857a8f854d662b688be6cfb54bb8636c9541a/src/Microsoft.PowerShell.Commands.Management/commands/management/GetContentCommand.cs#L358-L359>

Reply:
Thanks! I thought abot removing `ToArray()` but we can leave it.